cylc/rose/platform_utils.py

```diff
@@ -75,6 +81,15 @@ def get_platform_from_task_def(flow: str, task: str) -> Dict[str, Any]:
     return platform
 
 
+def force_compat_mode(flow_file):
+    """Check whether this is a Cylc 7 Back compatibility mode workflow:
+    https://github.com/cylc/cylc-rose/issues/319
+    """
+    return (
+        Path(flow_file).parent.parent.parent / WorkflowFiles.SUITE_RC
+    ).exists()
+
+
```

Comment:
Could simplify this by passing in the run directory rather than walking up the tree of parents to get there.

```suggestion
def check_compat_mode(run_dir):
    """Check whether this is a Cylc 7 Back compatibility mode workflow.
    
    See https://github.com/cylc/cylc-rose/issues/319
    
    Args:
        run_dir: Cylc workflow run directory.

    """
    return (run_dir / WorkflowFiles.SUITE_RC).exists()


```